### ISSUE TITLE
[v2.2.0-beta.1] vm.hasOwnProperty(someProp) issue

### ISSUE THREAD

**Issue by Leopoldthecoder** [vuejs/vue#5001]
### Vue.js version
2.2.0-beta.1

### Reproduction Link
http://jsfiddle.net/leopoldthecuber/z8oemLz3/

### Steps to reproduce
Just run the fiddle

### What is Expected?
Alerts `true`

### What is actually happening?
Alerts `false`

This is different from version 2.1.10: http://jsfiddle.net/leopoldthecuber/z8oemLz3/1/ , where `vm.hasOwnProperty('someProp') === true`. Perhaps a bug?

**Comment by yyx990803** [comment]
In 2.2 there's an optimization: all the getters for props and computed properties are defined on the component's prototype now. This avoids having to dynamically call `Object.defineProperty` for every property in every component instance and should result in faster initialization performance.

In practice I've never had to rely on `hasOwnProperty` checks in my components - is this commonly used in your projects?

**Comment by Leopoldthecoder** [comment]
Ah that makes sense.

I just encountered a [bug](http://jsfiddle.net/leopoldthecuber/n10p00s5/2/) and tracked down to this. In my code I did something like [this](http://jsfiddle.net/leopoldthecuber/z8oemLz3/). Since `name` is not the component's own property, it is missing in `props`.

This seems to be an extreme edge case, and in my particular case there are some workarounds. What is your opinion on this?
